src/transaction/flashback.c

```diff
@@ -64,9 +64,92 @@ static volatile int flashback_Threshold_to_remove_archive = 0;	/* If the differe
 								 * the archive log volume can be deleted.
 								 */
 // *INDENT-OFF*
-static std::atomic_bool flashback_Is_active = false;	// the status value that the flashback is processing the request
+static std::atomic_bool flashback_Is_active = false;	// the status value that the flashback is processing the a request
+static std::atomic_bool flashback_In_progress = false;  // the status value that the flashback is in progress
 // *INDENT-ON*
 
+static CSS_CONN_ENTRY *flashback_Current_conn = NULL;	// the connection entry for a flashback requests
+
+static pthread_mutex_t flashback_Request_lock = PTHREAD_MUTEX_INITIALIZER;
+
+/*
+ * flashback_is_duplicated_request - check if the caller is duplicated request for flashback
+ *
+ * return   : duplicated or not
+ */
+
+void
+flashback_lock_request ()
+{
+  pthread_mutex_lock (&flashback_Request_lock);
+}
+
+void
+flashback_unlock_request ()
+{
+  pthread_mutex_unlock (&flashback_Request_lock);
+}
+
+bool
+flashback_is_duplicated_request (THREAD_ENTRY * thread_p)
+{
+  CSS_CONN_ENTRY *previous_conn = NULL;
+
+  /* When flashback is first called or exited properly,
+   * flashback_Current_conn is set to NULL */
+  if (flashback_Current_conn == NULL)
+    {
+      return false;
+    }
+
+  /* flashback_Current_conn indicates conn_entry in thread_p, and conn_entry can be reused by request handler.
+   * So, status in flashback_Current_conn can be overwritten. */
+
+  previous_conn = flashback_Current_conn;
+
+  if (flashback_In_progress == false)
+    {
+      /* previous flashback set flashback_In_progress to false prolperly. (exited well) */
+      return false;
+    }
+  else
+    {
+      /* flashback_In_progress == true */
+
+      if (previous_conn->in_flashback == true)
```

Comment:
Yes, CSS_CONN_ENTRY is the one of the entries in css_Conn_array, which is allocated at server startup. 
And css_Conn_array is deallocated on server shutdown.

Or does it mean that the connection is lost in the process rather than deallocating the memory?

Reply:
No, I meant deallocating. Thank you!